### PR TITLE
Add price-capacity-optimized AllocationStrategy

### LINKED ISSUE #6058: Feature request:  price-capacity-optimized
We have been using Parallelcluster 2.x with SGE for quite some time, manually editing the autoscaling groups, and using the "price-capacity-optimized" spot allocation strategy introduced to ASGs in late 2022:  https://aws.amazon.com/blogs/compute/introducing-price-capacity-optimized-allocation-strategy-for-ec2-spot-instances/

We've more recently switched to Parallelcluster 3 with Slurm, and find that our only AllocationStrategy options are "lowest-price" or "capacity-optimized".  We'd like to be able to use price-capacity-optimized, but it's not an option:  
https://docs.aws.amazon.com/parallelcluster/latest/ug/Scheduling-v3.html#yaml-Scheduling-SlurmQueues-AllocationStrategy

I suspect many here may be very interested in having this AllocationStrategy available.

Thanks,
Dan

### PR DESCRIPTION
**Pull request by mflevine**
Resolves #6058

### CODE REVIEW

**Comment by mflevine** [comment]
Can a team member please review this.

File: cli/tests/pcluster/validators/test_instances_validators.py [hunk 1]
```diff
@@ -593,11 +593,18 @@ def test_instances_networking_validator(
...
         ),
+        (
+            CapacityType.ONDEMAND,
+            AllocationStrategy.PRICE_CAPACITY_OPTIMIZED,
+            "Compute Resource TestComputeResource is using an OnDemand CapacityType but the Allocation Strategy "
+            "specified is price-capacity-optimized. OnDemand CapacityType can only use 'lowest-price' allocation strategy.",
+        ),
         (CapacityType.ONDEMAND, AllocationStrategy.LOWEST_PRICE, ""),
         (CapacityType.ONDEMAND, None, ""),
         # Spot Capacity type supports both "lowest-price" and "capacity-optimized" allocation strategy
```
Comment (dreambeyondorange) [inline cli/tests/pcluster/validators/test_instances_validators.py:605]: nit: could you update the comment here?

**Comment by dreambeyondorange** [comment]
Thank you for the contribution!  Would it be possible for you to add an integration test or add this feature to an existing integration test and verify it's functionality as part of this PR?  That would help expedite inclusion in a future release.

Thanks!

**Comment by mflevine** [comment]
@dreambeyondorange I have added an integration test

**Comment by dreambeyondorange** [comment]
> @dreambeyondorange I have added an integration test

LGTM!  Can you just add an update to the Changelog to include the new feature?   Looks like there are a couple linter errors as well.  Once you add these I should be able to approve and merge!

**Comment by mflevine** [comment]
Updated!

**Review by dreambeyondorange — APPROVED** [review]

**Pull request merged**